### PR TITLE
Change consts to enums

### PR DESCRIPTION
**Pull request by jmythms**
Pull request overview
---------------------
Changing conts to enums.

### CODE REVIEW

**Comment by Myoldmopar** [comment]
So....why is CI still wanting to run `5ZoneAirCooledConvCoefPIU` in this branch?  That file was deleted in #8396.  Oh I see, the IDF was deleted but the CMakeLists entry was left in.  OK, I'm making that tweak in this branch locally and I'll push it up then CI will report back all green and this can go in.  Thanks @jmythms !

**Comment by jmythms** [comment]
Thank you very much!

**Comment by jmythms** [comment]
Messed up here, did not mean to push to this branch. Reverting commits.

**Comment by Myoldmopar** [comment]
@jmythms sorry for the confusion, the IDF actually should've stayed removed.  It was removed in a different PR but the entry in `testfiles/CMakeLists.txt` was accidentally left in.  The correct action should've just been to delete [this line](https://github.com/NREL/EnergyPlus/blob/1a0f1a7128599644c626261f880e89b1a9554e60/testfiles/CMakeLists.txt#L38).  I'll take care of it.

**Comment by jmythms** [comment]
Thank you for the help and for explaining it, Edwin. I'll check for it next time if it happens again.

File: .github/workflows/documentation.yml
Comment (Myoldmopar) [inline .github/workflows/documentation.yml]: This needs to be reverted back as well, I'm not sure how that happened.

File: src/EnergyPlus/OutputProcessor.hh
Comment (Myoldmopar) [inline src/EnergyPlus/OutputProcessor.hh:84]: Yay for constexprs!

File: src/EnergyPlus/PluginManager.cc
Comment (Myoldmopar) [inline src/EnergyPlus/PluginManager.cc]: Yeah something went wrong here, I don't know why these fixes are missing here.

File: testfiles/CMakeLists.txt
Comment (Myoldmopar) [inline testfiles/CMakeLists.txt:38]: This is good though.

**Review by Myoldmopar — COMMENTED** [review]
OK, this looks almost ready.  I'm not sure why a couple fixes were backed out here but I'll fix that and push.  If it's green after that this can go in.

**Review by Myoldmopar — APPROVED** [review]

**Comment by Myoldmopar** [comment]
I'll let CI have one final pass over these changes, but they look good locally.

**Comment by Myoldmopar** [comment]
All good here; this will merge in a few minutes when CI is done.  Thanks for this @jmythms.

**Comment by jmythms** [comment]
Thank you @Myoldmopar !

**Pull request merged**
